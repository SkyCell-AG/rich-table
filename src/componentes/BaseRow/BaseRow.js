import React, {
    useCallback,
    useState,
} from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import noop from 'lodash/noop'
import get from 'lodash/get'
import ChevronRight from '@material-ui/icons/ChevronRight'

import getComponentProps from 'utils/getComponentProps'

import BaseCell from 'componentes/BaseCell'

import useStyles from './BaseRow.style'

const propTypes = {
    name: PropTypes.string.isRequired,
    selectedRows: PropTypes.object, // eslint-disable-line
    rowSelected: PropTypes.string,
    uniqFieldValue: PropTypes.string.isRequired,
    rowClick: PropTypes.func.isRequired,
    rowProps: PropTypes.object, // eslint-disable-line
    columns: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            Header: PropTypes.func,
            Cell: PropTypes.func,
            props: PropTypes.object, // eslint-disable-line
        }).isRequired,
    ).isRequired,
    detailPanel: PropTypes.func,
    selectedRowId: PropTypes.string,
}

const defaultProps = {
    selectedRows: {},
    detailPanel: undefined,
    rowSelected: undefined,
    selectedRowId: undefined,
    rowProps: {},
}

const BaseRow = (props) => {
    const {
        name,
        selectedRows,
        rowSelected,
        selectedRowId,
        uniqFieldValue,
        rowClick,
        rowProps,
        columns,
        detailPanel: DetailPanel,
    } = props

    const classes = useStyles()

    const [
        isHide,
        setHide,
    ] = useState(true)
    const togglePanel = useCallback(() => {
        setHide(!isHide)
    }, [isHide])

    return (
        <div
            data-test-id="richtable-row"
            className={clsx({
                [classes.selectedRow]: selectedRowId === uniqFieldValue,
            })}
            onKeyDown={DetailPanel ? togglePanel : rowClick(rowProps)}
            onClick={DetailPanel ? togglePanel : rowClick(rowProps)}
        >
            <div
                className={clsx(
                    classes.rowContent,
                    {
                        [classes.cursorPointer]: !Array.isArray(selectedRows),
                        [classes.selectedRowContent]: rowSelected,
                    },
                )}
            >
                {
                    DetailPanel && (
                        <div className={classes.iconWrapper}>
                            <ChevronRight className={clsx(
                                {
                                    [classes.isOpenIcon]: !isHide,
                                },
                            )}
                            />
                        </div>
                    )
                }
                {
                    columns.map(({
                        Cell = BaseCell,
                        id,
                        width,
                        mapCellProps = noop,
                        component: Component,
                        propsMapper,
                        props: columnProps,
                    }) => {
                        const value = get(rowProps, mapCellProps)

                        const cellProps = typeof mapCellProps === 'string' ? {
                            children: Component
                                ? (
                                    <Component
                                        name={name}
                                        value={value}
                                        mapCellProps={mapCellProps}
                                        {...getComponentProps(
                                            propsMapper,
                                            rowProps,
                                        )}
                                    />
                                )
                                : value,
                        } : mapCellProps(rowProps)

                        return (
                            <div
                                style={{
                                    width,
                                }}
                                className={classes.cell}
                                key={`cell-${id}`}
                                data-testid={`cell-${id}`}
                            >
                                <Cell
                                    id={id}
                                    {...rowProps}
                                    {...columnProps}
                                    {...cellProps}
                                />
                            </div>
                        )
                    })
                }
            </div>
            {
                DetailPanel && (
                    <div className={clsx(
                        {
                            [classes.hideDetails]: isHide,
                        },
                    )}
                    >
                        <DetailPanel
                            name={name}
                            rowProps={rowProps}
                        />
                    </div>
                )
            }
        </div>
    )
}

BaseRow.propTypes = propTypes
BaseRow.defaultProps = defaultProps

export default BaseRow
