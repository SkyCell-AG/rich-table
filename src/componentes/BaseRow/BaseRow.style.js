import {
    makeStyles,
} from '@material-ui/core'

const useStyles = makeStyles((theme) => {
    return {
        selectedRow: {
            background: `${theme.palette.primary[50]} !important`,
        },
        rowContent: {
            display: 'flex',
            justifyContent: 'space-around',
            minHeight: '40px',
            '&:hover': {
                backgroundColor: theme.palette.primary[50],
            },
        },
        cursorPointer: {
            cursor: 'pointer',
        },
        selectedRowContent: {
            opacity: 0.7,
        },
        cell: {
            width: '100%',
            overflow: 'hidden',
            borderBottom: `1px solid ${theme.palette.secondary[100]}`,
        },
        hideDetails: {
            display: 'none',
        },
        showDetails: {
            display: 'block',
        },
        iconWrapper: {
            height: 'auto',
            borderBottom: `1px solid ${theme.palette.secondary[100]}`,
            display: 'flex',
            alignItems: 'center',
            color: theme.palette.secondary[600]
        },
        isOpenIcon: {
            transform: 'rotate(90deg)',
        }
    }
})

export default useStyles
