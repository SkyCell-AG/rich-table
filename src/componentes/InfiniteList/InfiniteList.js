import React from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import CircularProgress from '@mui/material/CircularProgress'
// import SnackbarContent from '@mui/material/SnackbarContent'

import {
    PENDING,
    FAILURE,
    SUCCESS,
    requestType,
} from 'utils/requestStatuses'

import useStyles from './InfiniteList.style'

const propTypes = {
    data: PropTypes.array, // eslint-disable-line
    status: requestType.isRequired,
    onScroll: PropTypes.func.isRequired,
    className: PropTypes.string,
    Row: PropTypes.func.isRequired,
    BeforeList: PropTypes.func,
}

const defaultProps = {
    data: [],
    className: '',
    BeforeList: null,
}

const InfiniteList = ({
    data,
    status,
    onScroll,
    wrapperRef, // eslint-disable-line
    spacerRef, // eslint-disable-line
    className,
    BeforeList,
    Row,
}) => {
    const classes = useStyles()

    return (
        <div
            onScroll={onScroll}
            ref={wrapperRef}
            className={clsx(classes.wrapper, className)}
            data-testid="infinitelist"
        >
            {BeforeList && <BeforeList data={data} />}
            {status === FAILURE && (
                <div className={classes.noDataContainer}>
                    <div className={classes.noData}>
                        Something went wrong and the data could not be loaded.
                    </div>
                </div>
            )}
            {SUCCESS === status && data.length === 0 && (
                <div className={classes.noDataContainer}>
                    <div className={classes.noData}>
                        There is no data in this table yet.
                    </div>
                </div>
            )}
            {data.map(Row)}
            {status === PENDING && (
                <CircularProgress
                    className={classes.loader}
                    size={40}
                />
            )}
            <div ref={spacerRef} />
        </div>
    )
}

InfiniteList.propTypes = propTypes
InfiniteList.defaultProps = defaultProps

export default InfiniteList
