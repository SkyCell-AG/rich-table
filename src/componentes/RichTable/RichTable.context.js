import {
    createContext,
} from 'react'
import noop from 'lodash/noop'

const RichTableContext = createContext({
    setDataTable: noop,
})

export default RichTableContext
