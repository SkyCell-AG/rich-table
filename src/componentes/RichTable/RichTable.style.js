import {
    makeStyles,
} from '@material-ui/styles'

const useStyles = makeStyles({
    headings: {
        position: 'sticky',
        zIndex: 1,
        top: 0,
        display: 'flex',
        justifyContent: 'space-around',
        boxShadow: (theme) => { return `0 1px 2px 0 ${theme.palette.secondary[100]}` },
        minHeight: (theme) => { return theme.shape.tabHeight },
        backgroundColor: (theme) => { return theme.palette.common.white },
    },
    tableWrapper: {
        position: 'relative',
        width: '100%',
        display: 'flex',
        flexDirection: 'column',
        fontSize: '1rem',
    },
    tableWrapperWhenEditing: {
        marginRight: 0,
    },
    tabs: {
        display: 'flex',
        alignItems: 'center',
    },
    headerTableWrapper: {
        display: 'flex',
        position: 'absolute',
        width: '100%',
        overflow: 'hidden',
        overflowX: 'auto',
        justifyContent: 'space-between',
    },
    controlsWrapper: {
        display: 'flex',
        justifyContent: 'right',
        maxHeight: (theme) => { return theme.shape.tabHeight },
    },
    contentWrapperWhenEditing: {
        borderRight: 'none !important',
    },
    controls: {
        display: 'flex',
        zIndex: 1,
        height: (theme) => { return theme.shape.tabHeight },
        backgroundColor: (theme) => { return theme.palette.secondary[50] },
        paddingLeft: 10,
        whiteSpace: 'nowrap',
        borderRadius: '10px 10px 0px 0px',
        borderWidth: 1,
        borderStyle: 'solid',
        borderColor: (theme) => { return theme.palette.secondary[300] },
        borderBottom: (theme) => { return `1px solid ${theme.palette.secondary[50]}` },
    },
    control: {
        cursor: 'pointer',
        display: 'flex',
        alignItems: 'center',
        color: (theme) => { return theme.palette.secondary[600] },
        padding: '2px 20px 2px 0px',
    },
    contentWrapper: {
        position: 'relative',
        left: 0,
        padding: (theme) => { return theme.spacing(1.5) },
        top: (theme) => { return theme.shape.tabHeight - 1 },
        width: (theme) => { return `calc(100% - ${theme.spacing(3)}px)` },
        height: (theme) => { return `calc(100% - ${theme.shape.tabHeight + theme.spacing(3) + 2}px)` },
        borderRadius: '10px 0 0 0',
        backgroundColor: (theme) => { return theme.palette.secondary[50] },
        borderWidth: 1,
        borderStyle: 'solid',
        borderColor: (theme) => { return theme.palette.secondary[300] },
    },
    cell: {
        width: '100%',
        overflow: 'hidden',
    },
    cursorPointer: {
        cursor: 'pointer',
    },
    withDetailPanel: {
        paddingLeft: 24,
    },
})

export default useStyles
