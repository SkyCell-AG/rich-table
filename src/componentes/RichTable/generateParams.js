import memoize from 'lodash/memoize'
import isEmpty from 'lodash/isEmpty'

const generateParams = memoize((filter, sort, excludeFilters) => {
    const params = {}

    if (!isEmpty(filter)) {
        params.filter = filter
    }

    if (!isEmpty(sort)) {
        params.sort = sort
    }
    if (!isEmpty(excludeFilters)) {
        params.excludeFilters = excludeFilters
    }

    return params
}, (...data) => {
    return JSON.stringify(data)
})

export default generateParams
