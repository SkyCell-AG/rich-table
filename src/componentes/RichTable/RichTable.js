import React, {
    useCallback,
    forwardRef,
    useEffect,
    useState,
    useRef,
} from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import noop from 'lodash/noop'
import get from 'lodash/get'

import HeaderCell from 'componentes/HeaderCell'
import InfiniteList from 'componentes/InfiniteList'
import DndCell from 'componentes/DndCell'
import BaseRow from 'componentes/BaseRow'

import useStyles from './RichTable.style'

const propTypes = {
    uniqField: PropTypes.string,
    name: PropTypes.string.isRequired,
    configName: PropTypes.string,
    load: PropTypes.func.isRequired,
    visible: PropTypes.arrayOf(PropTypes.string).isRequired,
    setVisible: PropTypes.func.isRequired,
    onRowClick: PropTypes.func,
    changeSequence: PropTypes.func.isRequired,
    setMatchedResults: PropTypes.func.isRequired,
    matchedResults: PropTypes.number,
    removeFilter: PropTypes.func.isRequired,
    removeSort: PropTypes.func.isRequired,
    rerenderInfinitList: PropTypes.func.isRequired,
    infinitListKey: PropTypes.string.isRequired,
    selectedRows: PropTypes.arrayOf(
        PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
        ]),
    ),
    columns: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            Header: PropTypes.func,
            Cell: PropTypes.func,
            props: PropTypes.object, // eslint-disable-line
        }).isRequired,
    ).isRequired,
    className: PropTypes.string,
    controlPanel: PropTypes.func,
    renderControlPanel: PropTypes.func,
    classNames: PropTypes.exact({
        root: PropTypes.string,
        contentWrapper: PropTypes.string,
        headerWrapper: PropTypes.string,
    }),
    editing: PropTypes.bool,
    selectedRowId: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
    detailPanel: PropTypes.func,
    openRowId: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
    rowStyle: PropTypes.shape({
        crossLine: PropTypes.string,
    }),
    disabled: PropTypes.bool,
}

const defaultProps = {
    matchedResults: undefined,
    className: '',
    onRowClick: undefined,
    selectedRows: undefined,
    uniqField: 'id',
    controlPanel: null,
    configName: undefined,
    renderControlPanel: null,
    classNames: {},
    editing: false,
    selectedRowId: '',
    detailPanel: undefined,
    openRowId: undefined,
    rowStyle: undefined,
    disabled: false,
}

const RichTable = forwardRef((props, ref) => {
    const {
        columns,
        className: wrapperClassName,
        setMatchedResults,
        onRowClick,
        selectedRows,
        infinitListKey,
        changeSequence,
        uniqField,
        controlPanel: ControlPanel,
        renderControlPanel,
        classNames,
        editing,
        selectedRowId,
        name,
        configName,
        detailPanel,
        rerenderInfinitList,
        openRowId,
        load,
        rowStyle,
        disabled,
    } = props

    const classes = useStyles()

    const [
        openedRow,
        setOpenedRow,
    ] = useState(null)

    const closeOpenedRow = useCallback(() => {
        setOpenedRow(null)
    }, [])

    const rowClick = useCallback((row) => {
        return () => {
            if (disabled) { return }
            if (onRowClick) {
                onRowClick(
                    row,
                    rerenderInfinitList,
                )
            } else if (detailPanel) {
                setOpenedRow(openedRow === row ? null : row)
            }
        }
    }, [
        openedRow,
        onRowClick,
        detailPanel,
        rerenderInfinitList,
        disabled,
    ])

    const selectedElm = useRef()

    useEffect(() => {
        if (selectedRowId && selectedElm.current) {
            if (selectedElm.current.scrollIntoViewIfNeeded) {
                selectedElm.current.scrollIntoViewIfNeeded()
            } else {
                // IE, Firefox
                selectedElm.current.scrollIntoView()
            }
        }
    }, [selectedRowId])

    const renderedControlPanel = (renderControlPanel) ? renderControlPanel(props) : null

    return (
        <div
            className={clsx(
                classNames.root,
                classes.tableWrapper,
                editing && classes.tableWrapperWhenEditing,
                wrapperClassName,
            )}
        >
            <div
                className={clsx(
                    classes.headerTableWrapper,
                    editing && classes.tableWrapperWhenEditing,
                    classNames.headerWrapper,
                )}
            >
                {ControlPanel && (
                    <ControlPanel
                        {...props}
                    />
                )}
                {renderedControlPanel}
            </div>

            <div
                className={clsx(
                    classes.contentWrapper,
                    editing && classes.contentWrapperWhenEditing,
                    classNames.contentWrapper,
                )}
            >
                <InfiniteList
                    {...props}
                    load={load}
                    ref={ref}
                    key={infinitListKey}
                    onUpdateMatchedResults={setMatchedResults}
                    BeforeList={({
                        data,
                    }) => {
                        return (
                            <div className={clsx(
                                classes.headings,
                                {
                                    [classes.withDetailPanel]: detailPanel,
                                },
                            )}
                            >
                                {columns.map(({
                                    Header = HeaderCell,
                                    mapHeaderProps = noop,
                                    headerProps: headerPropsOut,
                                    filterField,
                                    id,
                                    className: cellClassName,
                                    width,
                                    minWidth = '',
                                    props: columnProps,
                                    ...rest
                                }) => {
                                    const headerProps = headerPropsOut
                                        || mapHeaderProps({
                                            ...columnProps,
                                            data,
                                        })

                                    const divStyles = width ? {
                                        flex: `0 0 ${width}`,
                                        minWidth: `${minWidth}`,
                                    } : {
                                        minWidth: `${minWidth}`,
                                    }

                                    return (
                                        <div
                                            key={`DndCell-${id}`}
                                            data-testid={`table-header-${id}`}
                                            className={clsx(
                                                classes.cell,
                                                cellClassName,
                                            )}
                                            style={divStyles}
                                        >
                                            <DndCell
                                                {...headerProps}
                                                onDrop={changeSequence}
                                                id={id}
                                                type="HeaderCell"
                                                data={data}
                                            >
                                                <Header
                                                    {...rest}
                                                    {...columnProps}
                                                    {...headerProps}
                                                    data={data}
                                                    key={`header-${id}`}
                                                    id={filterField || id}
                                                />
                                            </DndCell>
                                        </div>
                                    )
                                })}
                            </div>
                        )
                    }}
                    Row={(rowProps) => {
                        const uniqFieldValue = get(rowProps, uniqField)

                        const innerOpenRowId = get(openedRow, uniqField)

                        const openRow = (
                            openRowId === uniqFieldValue
                            || innerOpenRowId === uniqFieldValue
                        )

                        return (
                            <div
                                ref={selectedRowId === uniqFieldValue ? selectedElm : null}
                                key={`row-${uniqFieldValue}`}
                            >
                                <BaseRow
                                    name={name}
                                    configName={configName}
                                    selectedRows={selectedRows}
                                    selectedRowId={selectedRowId}
                                    uniqFieldValue={uniqFieldValue}
                                    rowClick={rowClick(rowProps)}
                                    rowProps={rowProps}
                                    columns={columns}
                                    detailPanel={detailPanel}
                                    openRow={openRow}
                                    closeOpenedRow={closeOpenedRow}
                                    rowStyle={rowStyle}
                                    disabled={disabled}
                                />
                            </div>
                        )
                    }}
                />
            </div>
        </div>
    )
})

RichTable.propTypes = propTypes
RichTable.defaultProps = defaultProps

export default RichTable
