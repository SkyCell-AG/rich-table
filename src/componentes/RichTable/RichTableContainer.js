import React, {
    useReducer,
    useCallback,
    useMemo,
    useEffect,
} from 'react'
import PropTypes from 'prop-types'
import omit from 'lodash/omit'
import {
    v1 as uuid,
} from 'uuid'

import generateFilters from 'utils/generateFilters'
import reducer from '/store/reducer'
import * as statuses from 'utils/requestStatuses'
import generateParams from 'utils/generateParams'

import useLoadPage from 'hooks/useLoadPage'
import useVisible from 'hooks/useVisible'
import useSequence from 'hooks/useSequence'
import useFilter from 'hooks/useFilter'
import useSort from 'hooks/useSort'
import useSelectRow from 'hooks/useSelectRow'

import RichTable from './RichTable'

const initialState = {
    status: statuses.PRISTIN,
    data: [],
    matchedResults: 0,
    infinitListKey: uuid(),
}

const propTypes = {
    uniqField: PropTypes.string,
    visible: PropTypes.arrayOf(PropTypes.string),
    selectedRows: PropTypes.arrayOf(PropTypes.string),
    onSelectRow: PropTypes.func,
    load: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    onFilterChange: PropTypes.func,
    selectedFilter: PropTypes.object, // eslint-disable-line
    filter: PropTypes.object, // eslint-disable-line
    sort: PropTypes.object, // eslint-disable-line
    columns: PropTypes.arrayOf(PropTypes.shape({
        props: PropTypes.object, // eslint-disable-line
    })).isRequired,
}

const defaultProps = {
    sort: {},
    visible: [],
    filter: {},
    uniqField: 'id',
    onSelectRow: undefined,
    onFilterChange: undefined,
    selectedRows: undefined,
}

const RichTableContainer = ({
    columns: outColumns,
    load,
    name,
    selectedRows,
    visible: predefinedVisible,
    sort: predefinedSort,
    filter: predefinedFilter,
    onSelectRow,
    uniqField,
    namedQuery,
    ...props
}) => {
    const [
        state,
        dispatch,
    ] = useReducer(reducer, initialState)

    const  {
        infinitListKey,
        matchedResults,
    } = state

    const typeMapping = useMemo(() => {
        return outColumns.reduce((prev, {
            filterField,
            filterType,
        }) => {
            if (!filterField) {
                return prev
            }

            return {
                ...prev,
                [filterField]: filterType,
            }
        }, {})
    }, [outColumns])

    const columns = useMemo(() => {
        return outColumns.map((column) => {
            const {
                mapHeaderProps,
                headerProps,
            } = column

            if (headerProps) {
                return {
                    ...column,
                    label: headerProps.children,
                }
            }

            if (mapHeaderProps) {
                return {
                    ...column,
                    label: mapHeaderProps().children,
                }
            }

            return {
                ...column,
                label: '',
            }
        })
    }, [outColumns])

    const [
        sequence,
        changeSequence,
    ] = useSequence(columns)
    const [
        visible,
        setVisible,
    ] = useVisible(columns, predefinedVisible)
    const [
        filter,
        setFilter,
        removeFilter,
    ] = useFilter(predefinedFilter)
    const [
        sort,
        setSorting,
        removeSort,
    ] = useSort(predefinedSort)

    const params = useMemo(() => {
        return generateParams(filter, sort)
    }, [
        filter,
        sort,
    ])

    const loadWithParams = useCallback((page) => {
        return load({
            ...params,
            typeMapping,
            page,
        }).catch((err) => {
            // eslint-disable-next-line no-console, max-len
            console.error(err) // TODO: without catch handler catch handler in InfinitListContainer doesn't fire
        })
    }, [
        load,
        params,
        typeMapping,
    ])

    const {
        loadPage,
        handleSetMatchedResults,
        handleRerenderInfinitList,
    } = useLoadPage(loadWithParams, dispatch)

    const createFilterHandler = useCallback((columnId) => {
        return (value) => {
            setFilter(columnId, value)
        }
    }, [setFilter])

    const visibleAndSortedColumns = sequence
        .filter((columnId) => {
            return visible.includes(columnId)
        })
        .map((columnId) => {
            return columns.find(({
                id,
            }) => {
                return id === columnId
            })
        })
        .filter(Boolean)
        .map((column) => {
            const appliedFilters = generateFilters(omit(
                filter,
                [column.filterField],
            ), typeMapping)

            return {
                ...column,
                props: {
                    appliedFilters,
                    filter: filter[column.filterField],
                    predefinedFilter,
                    setFilter: createFilterHandler(column.filterField || column.id),
                    [sort.field === column.id && 'sortDirection']: sort.direction,
                    sort: (direction) => {
                        setSorting(
                            column.id,
                            direction,
                        )
                    },
                },
            }
        })

    const visibleAndSortedColumnsWithSelection = useSelectRow({
        columns: visibleAndSortedColumns,
        onSelectRow,
        uniqField,
    })

    useEffect(() => {
        loadPage([], 1)
    }, [filter, namedQuery])

    return (
        <RichTable
            {...props}
            {...state}
            name={name}
            load={loadPage}
            filter={filter}
            removeFilter={removeFilter}
            rerenderInfinitList={handleRerenderInfinitList}
            removeSort={removeSort}
            setMatchedResults={handleSetMatchedResults}
            changeSequence={changeSequence}
            visible={visible}
            setVisible={setVisible}
            allColumns={columns}
            infinitListKey={infinitListKey}
            matchedResults={matchedResults}
            selectedRows={selectedRows}
            columns={visibleAndSortedColumnsWithSelection}
            uniqField={uniqField}
        />
    )
}

RichTableContainer.propTypes = propTypes
RichTableContainer.defaultProps = defaultProps

export default RichTableContainer
