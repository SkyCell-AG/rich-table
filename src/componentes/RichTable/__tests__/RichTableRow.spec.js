import React from 'react'
import {
    shallow,
} from 'enzyme'

import InfiniteList from 'componentes/InfiniteList'
import BaseRow from 'componentes/BaseRow'

import RichTable from '../RichTable'

const rowProps = {
    id: 'some-id',
    column1: 'column1',
    column2: 'column2',
}

describe('Richtable Row', () => {
    test('snapshot', () => {
        const element = shallow(
            <RichTable
                name="someName"
                load={jest.fn()}
                rerenderInfinitList={jest.fn()}
                changeSequence={jest.fn()}
                removeSort={jest.fn()}
                removeFilter={jest.fn()}
                visible={[
                    'column1',
                    'column2',
                ]}
                infinitListKey="infinitListKey"
                setVisible={jest.fn()}
                columns={[
                    {
                        id: 'column1',
                        props: {
                            column1: 'column1',
                        },
                    },
                    {
                        id: 'column2',
                        props: {
                            column2: 'column2',
                        },
                    },

                ]}
                setMatchedResults={jest.fn()}
            />,
        )

        const RowComponent = element.find(InfiniteList)
            .props()
            .Row

        expect(
            shallow(
                <RowComponent {...rowProps} />,
            ),
        ).toMatchSnapshot()
    })

    test('row select and unselect', () => {
        const element = shallow(
            <RichTable
                name="someName"
                load={jest.fn()}
                rerenderInfinitList={jest.fn()}
                changeSequence={jest.fn()}
                removeSort={jest.fn()}
                removeFilter={jest.fn()}
                visible={[
                    'column1',
                    'column2',
                ]}
                detailPanel={() => {
                    return <div>DetailPanel</div>
                }}
                infinitListKey="infinitListKey"
                setVisible={jest.fn()}
                columns={[
                    {
                        id: 'column1',
                        props: {
                            column1: 'column1',
                        },
                    },
                    {
                        id: 'column2',
                        props: {
                            column2: 'column2',
                        },
                    },

                ]}
                setMatchedResults={jest.fn()}
            />,
        )

        const RowComponent = element.find(InfiniteList)
            .props()
            .Row

        const rowElement = shallow(
            <RowComponent
                {...rowProps}
            />,
        ).find(BaseRow)

        expect(
            rowElement.props().openRow,
        ).toBeFalsy()

        rowElement.props().rowClick()

        const RowComponentOpened = element.find(InfiniteList)
            .props()
            .Row

        const rowElementOpened = shallow(
            <RowComponentOpened
                id="some-id"
                column1="column1"
                column2="column2"
            />,
        ).find(BaseRow)

        expect(
            rowElementOpened.props().openRow,
        ).toBeTruthy()

        rowElementOpened.props().closeOpenedRow()

        const RowComponentClosed = element.find(InfiniteList)
            .props(
            )
            .Row

        const rowElementClosed = shallow(
            <RowComponentClosed
                {...rowProps}
            />,
        ).find(BaseRow)

        expect(
            rowElementClosed.props().openRow,
        )
            .toBeFalsy()
    })
})
