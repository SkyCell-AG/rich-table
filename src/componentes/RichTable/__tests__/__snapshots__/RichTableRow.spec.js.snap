// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Richtable Row snapshot 1`] = `
<div
  key="row-some-id"
>
  <BaseRow
    closeOpenedRow={[Function]}
    columns={
      Array [
        Object {
          "id": "column1",
          "props": Object {
            "column1": "column1",
          },
        },
        Object {
          "id": "column2",
          "props": Object {
            "column2": "column2",
          },
        },
      ]
    }
    name="someName"
    openRow={false}
    rowClick={[Function]}
    rowProps={
      Object {
        "column1": "column1",
        "column2": "column2",
        "id": "some-id",
      }
    }
    selectedRowId=""
    uniqFieldValue="some-id"
  />
</div>
`;
